Added release notes for 4.3.0 (#13923)
docker/docker.github.io: desktop/mac/release-notes/index.md

Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/projects/1){: target="_blank" rel="noopener" class="_"} to see what's coming next.

## Docker Desktop 4.3.0
2021-12-02

> Download Docker Desktop
>
> [Mac with Intel chip](https://desktop.docker.com/mac/main/amd64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-amd64){: .button .primary-btn }
> [Mac with Apple chip](https://desktop.docker.com/mac/main/arm64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-arm64){: .button .primary-btn }

### Upgrades

- [Docker Engine v20.10.11](https://docs.docker.com/engine/release-notes/#201011)
- [containerd v1.4.12](https://github.com/containerd/containerd/releases/tag/v1.4.12)
- [Buildx 0.7.1](https://github.com/docker/buildx/releases/tag/v0.7.1)

> - [Compose v2.2.1](https://github.com/docker/compose/releases/tag/v2.2.1)

Reviewer: I think we may want to elaborate a bit for the components we own (Engine, Buildx, Compose, …) with a sub-list of changes? Or is this something we’ve decided not to do in the end @usha-mandya?

Reviewer 2: I don't think it was intentional. We can add a list highlighting some important updates as per https://docs.docker.com/desktop/mac/release-notes/#docker-desktop-401.

> - [Go 1.17.3](https://golang.org/doc/go1.17)

Reviewer: Move Go to the bottom of the list, maybe? Most users don’t care about that.

- [Kubernetes 1.22.4](https://github.com/kubernetes/kubernetes/releases/tag/v1.22.4)

> - [hub-tool 0.4.4](https://github.com/docker/hub-tool/releases/tag/v0.4.4)

Reviewer: ```suggestion
- [Docker Hub Tool v0.4.4](https://github.com/docker/hub-tool/releases/tag/v0.4.4)
```
See e.g. https://docs.docker.com/desktop/windows/release-notes/3.x/#upgrades-3

> ### Bug fixes and minor changes

Reviewer 2: @trungutt Added review comments in the Windows section. PTAL and update the common ones here

> - Improved: Docker Desktop on Apple Silicon no longer requires Rosetta 2.

Reviewer 2: ```suggestion
- Improved: Docker Desktop on Apple silicon no longer requires Rosetta 2.
```

> - Improved: Docker Desktop on Apple silicon no longer requires Rosetta 2.

Reviewer: Does this need the prefix?
Why not just
```suggestion
- Docker Desktop on Apple silicon no longer requires Rosetta 2.
```

> - Fixed an issue using the "Save As" action in the Volumes UI to save files from a volume. Fixes [docker/for-win#12407](https://github.com/docker/for-win/issues/12407).

Reviewer: Maybe we could briefly explain what was the issue? Most users don’t click links but only quickly scan the changes.

Reviewer 2: Fixed an issue which prevented users from saving files from a volume using the **Save As** option in the Volumes UI.

> - Fixed applying an update after downloading it was sometimes re-launching the same version instead of the new one.

Reviewer 2: ```suggestion
- Fixed an issue that sometimes caused Docker Desktop to launch the existing version instead of a new version after downloading the update.
```

Reviewer 2: @trungutt How about "Fixed an issue that sometimes launched the existing version of Docker Desktop even after updating to a newer version."

Author: Thanks a lot

Reviewer: Should we point out that the fix will only take effect for the next update? 🤔

> - Fixed an issue that caused Docker Desktop to fail during startup if the home directory path contains a character used in regular expressions. Fixes [docker/for-win#12374](https://github.com/docker/for-win/issues/12374).

Reviewer: That one is only Windows, we don’t want it in the Mac release notes.

- Added a self-diagnose warning if the host lacks Internet connectivity.
- Docker Desktop now uses cgroupv2. If you need to run `systemd` in a container then:
  - Ensure your version of `systemd` supports cgroupv2. [It must be at least `systemd` 247](https://github.com/systemd/systemd/issues/19760#issuecomment-851565075). Consider upgrading any `centos:7` images to `centos:8`.
  - Containers running `systemd` need the following options: [`--privileged --cgroupns=host -v /sys/fs/cgroup:/sys/fs/cgroup:rw`](https://serverfault.com/questions/1053187/systemd-fails-to-run-in-a-docker-container-when-using-cgroupv2-cgroupns-priva).
- Deprecated: The following internal DNS names are deprecated and will be removed from a future release: `docker-for-desktop`, `docker-desktop`, `docker.for.mac.host.internal`, `docker.for.mac.localhost`, `docker.for.mac.gateway.internal`. You must now use `host.docker.internal`, `vm.docker.internal`, and `gateway.docker.internal`.

> - Removed: Custom RBAC rules have been removed from Docker Desktop as it gives `cluster-admin` privileges to all Service Accounts. Fixes [docker/for-mac/#4774](https://github.com/docker/for-mac/issues/4774).

Reviewer: Could these be moved to a «Deprecation» section, see https://docs.docker.com/desktop/windows/release-notes/3.x/#deprecation?

> 

Reviewer: ```suggestion
```
